RecoTracker/LST/plugins/alpaka/LSTInputProducer.cc

```diff
@@ -0,0 +1,225 @@
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESGetToken.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/TrackerRecHit2D/interface/Phase2TrackerRecHit1D.h"
+
+#include "FWCore/Utilities/interface/transform.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripMatchedRecHit2DCollection.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/TrackReco/interface/trackFromSeedFitFailed.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHitBuilder.h"
+
+#include "RecoTracker/LSTCore/interface/LSTInputHostCollection.h"
+#include "RecoTracker/LSTCore/interface/LSTPrepareInput.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class LSTInputProducer : public global::EDProducer<> {
+  public:
+    LSTInputProducer(edm::ParameterSet const& iConfig);
+    ~LSTInputProducer() override = default;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+    void produce(edm::StreamID, device::Event& iEvent, const device::EventSetup& iSetup) const override;
+
+    const double ptCut_;
+
+    const edm::EDGetTokenT<Phase2TrackerRecHit1DCollectionNew> phase2OTRecHitToken_;
+
+    const edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> mfToken_;
+    const edm::EDGetTokenT<reco::BeamSpot> beamSpotToken_;
+    std::vector<edm::EDGetTokenT<edm::View<reco::Track>>> seedTokens_;
+    const edm::EDPutTokenT<TrajectorySeedCollection> lstPixelSeedsPutToken_;
+
+    const edm::EDPutTokenT<lst::LSTInputHostCollection> lstInputPutToken_;
+  };
+
+  LSTInputProducer::LSTInputProducer(edm::ParameterSet const& iConfig)
+      : EDProducer<>(iConfig),
+        ptCut_(iConfig.getParameter<double>("ptCut")),
+        phase2OTRecHitToken_(consumes(iConfig.getParameter<edm::InputTag>("phase2OTRecHits"))),
+        mfToken_(esConsumes()),
+        beamSpotToken_(consumes(iConfig.getParameter<edm::InputTag>("beamSpot"))),
+        lstPixelSeedsPutToken_(produces()),
+        lstInputPutToken_(produces()) {
+    seedTokens_ =
+        edm::vector_transform(iConfig.getParameter<std::vector<edm::InputTag>>("seedTracks"),
+                              [&](const edm::InputTag& tag) { return consumes<edm::View<reco::Track>>(tag); });
```

Comment:
This could be done as part of the initializer list, in which case `seedTokens_` could be made `const`.